src/app.rs

```diff
@@ -44,6 +50,53 @@ impl EwwWindow {
     pub fn close(self) {
         self.gtk_window.close();
     }
+
+    #[cfg(not(target_os = "macos"))]
+    pub fn set_struts(&self) -> Result<()> {
```

Comment:
This should probably not be a method of `EwwWindow`, but instead be another free function that _takes_ `EwwWindow`, and is then used within `initialize_window`. Otherwise it would imply that it could make sense to call this multiple times, which really it doesn't - as setting struts is only ever part of the initial window initialization